### ISSUE TITLE
Wrong BundleIdentifier being used during “generating provisioningProfiles node”

### ISSUE THREAD

**Issue by bramus** [bitrise-steplib/steps-xcode-archive#136]
See https://discuss.bitrise.io/t/xcode-archive-export-for-ios-failing-with-name-app-requires-a-provisioning-profile-add-a-profile-to-the-provisioningprofiles-dictionary-in-your-export-options-property-list/6821 for an introduction.

In short: the generated `export_options.plist` file which is used for exporting the `.ipa` contained the wrong Bundle Identifier. This is most likely due to one of my earlier build steps, in which I manipulate the `CFBundleIdentifier` inside the app's `Info.plist` file (using `PlistBuddy`).

I don't know why exactly the “old” identifier is being used _(is it injected from somewhere?)_ but a possible fix could be for the `xcode-archive` to (re-)read the `CFBundleIdentifier` from the  `Info.plist` file just before the “generating provisioningProfiles node” part is being ran.

**Comment by bramus** [comment]
A bit of a late reply, but here goes:

I have not resolved this issue, but I have circumvented it.

What happens:

1. Build pipeline starts
2. A script in the build pipeline adjusts the `CFBundleIdentifier` on the fly, so that I can generate different builds. That way I can install these builds in parallel on my device.
3. The built `.xcarchive` cannot be exported to a `.ipa` (e.g. signed) because the generated `export_options.plist` file (used for signing) contains the “old" `CFBundleIdentifier` — e.g. it does not pick up the changes I did in step 2 described above.

How I circumvented it: I added a wildcard provisioning profile so that signing in step 3 does not complain.

Whilst this wildcard provisioning profile makes it all work I expected the changes made in step 2 to be take into account in step 3.

**Comment by bramus** [comment]
Still an issue most likely …

**Comment by mefiblogger** [comment]
Hello,

It's Mefi here from the Bitrise CoreSteps Team. Thanks for reaching out helping us and the mobile community to have a better experience with Bitrise steps. 💜

We started to have a broader review of all the open issues and PRs and we are continuously prioritizing these in our sprints.

We reviewed this issue and we found it to be a valid bug. We added this ticket to our internal backlog and we'll get back to you once we released the new version with the fix.

Happy building! 🚀

**Comment by lpusok** [comment]
Hello @bramus, sorry for the delay.
Are you (were you) explicitely specifying a Custom export options plist (via Step input)? That would prevent looking up-to-date bundle identifier. If not, would be grateful if you can provide a build URL to look into this.

Example log exrcept using custom export options:
```
Custom export options content provided, using it:
```

**Comment by lpusok** [comment]
Closing this thread due to inactivity. Please open a new issue if needed.
